### DISCUSSION TITLE
so-import-pcap missing infromation from pcaps

### DISCUSSION THREAD

**Discussion by gbrodt** [Security-Onion-Solutions/securityonion#7786] [Unsupported Versions] [answered]
We are using 2.3.100 in student pods for analysis and after importing a pcap file we have mixed results between the different pods of equipment. Alerts do not seem to generate. All imported the same file (malware traffic  analysis) yet the alerts and hunt information shows up differently.  All of the machines came from the same image and were duplicated to students. They are stand alone installs.

After upgrading to 2.3.110 today,  I have found if you use the URL shown after the import with the import.id you can find the information.  If you hunt by just the dates (with the correct utc time coverage) the events/traffic does not show.  

Also having issues with so-tcpreplay  /opt/so/samples/mysample.pcap   not showing up in the alerts either.

**Comment by gbrodt** [comment]
I do not see the alerts like at the start of the video: https://www.youtube.com/watch?v=dF2zWBO-Dgc
so-test is working fine.

**Comment by dougburks** [comment]
What are the specs of your student pods?
https://docs.securityonion.net/en/2.3/hardware.html#minimum-specs

When you go to the Alerts page, it defaults to the last 24 hours. Are you changing this default to include the time of the imported traffic?

When you go to Hunt, what query are you running?

Are you able to share screenshots?

**Comment by gbrodt** [comment]
Hi Doug,

TH security onion is running in a vm in the students pods (kali, windows, metaspolitable, pfsense, and some other on a LAN)
The SO has 6 cores and 20GB of RAM in it's VM in the pod Hosted on esxi 6.7.  Storage is 250GB thin provisioned for the VM.   The so-test works fine, I see the same issue with so-tcpreplay.

Yes, we are adjusting for the date and time (and UTC). After the SO import of the pcap the link at the command line does show the events I the hunt interface.

Here is a fresh import after a so-nsm-clear and a so-elastic-clear. There are two files from malware traffic analysis 8/19/2021 (funky lizard) and 03/21/2022 Brazil Malspam
(also removed them from /nsm/import)(so-status shows all OK too)

***@***.***

***@***.***


Alerts after importing these 2 files
***@***.***

If I look at alerts over valid timeframe I do not see any:
***@***.***


If I open the links give in the shell(is there a way to recall the links(would be nice if you re import the link would be shown ??) I get:
***@***.***
***@***.***


***@***.***


***@***.***


If I use a hunt "*" over the correct date timeframe they also show.
I can get you access to the pod I'm using if you want to see it firsthand.


Greg


From: Doug Burks ***@***.***>


Sent: Friday, April 15, 2022 6:26 AM
To: Security-Onion-Solutions/securityonion ***@***.***>
Cc: Brodt, Greg ***@***.***>; Author ***@***.***>
Subject: Re: [Security-Onion-Solutions/securityonion] so-import-pcap missing infromation from pcaps (Discussion #7786)


What are the specs of your student pods?
https://docs.securityonion.net/en/2.3/hardware.html#minimum-specs<https://nam04.safelinks.protection.outlook.com/?url=https%3A%2F%2Fdocs.securityonion.net%2Fen%2F2.3%2Fhardware.html%23minimum-specs&data=04%7C01%7CGreg.Brodt%40northwoodtech.edu%7C43d2495f8fbf4e02c35108da1ed2ac02%7Ca22a4ff1892a44ee8725a5114f07c7ce%7C0%7C0%7C637856187447526911%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000&sdata=9dAFuJCPmZehJXA7WN6AAg7ZGzxKZoLhppIp%2FA6ldCo%3D&reserved=0>

When you go to the Alerts page, it defaults to the last 24 hours. Are you changing this default to include the time of the imported traffic?

When you go to Hunt, what query are you running?

Are you able to share screenshots?

-

**Comment by gbrodt** [comment]
![Pensession1-000229](https://user-images.githubusercontent.com/75701987/163873799-41c0e2c8-ef7a-4a3d-9225-6bc51c6096e0.jpg)
![Pensession1-000230](https://user-images.githubusercontent.com/75701987/163873827-6d13df6c-e828-4d04-a63f-ecc961fa8f6b.jpg)
![Pensession1-000231](https://user-images.githubusercontent.com/75701987/163873834-be47600e-3fb1-464b-961d-274f8206cac4.jpg)
![Pensession1-000232](https://user-images.githubusercontent.com/75701987/163873839-af3ee48d-e760-49d2-8ac2-66a50b2b861a.jpg)
![Pensession1-000233](https://user-images.githubusercontent.com/75701987/163873843-ee988442-f0e1-4f3b-8f98-7d29dc226c07.jpg)
![Pensession1-000234](https://user-images.githubusercontent.com/75701987/163873853-da27aa28-ec72-447b-91a7-3e0653b37f38.jpg)
![Pensession1-000236](https://user-images.githubusercontent.com/75701987/163873854-432c7d54-fd00-45aa-98a0-08b850d3dd8c.jpg)
![Pensession1-000235](https://user-images.githubusercontent.com/75701987/163873865-cfe78712-3ca7-45da-861c-166715b7e6a9.jpg)
![Pensession1-000239](https://user-images.githubusercontent.com/75701987/163873872-2c16896d-fd82-4f6b-9396-d16de6c94442.jpg)

**Comment by dougburks** [comment]
Your Hunt screenshots show that all of the logs are Zeek logs and there are no NIDS alerts.

Do your machines have full Internet access?

Do they have a full NIDS ruleset?

What is the output of the following?
```
grep -v "^#" /opt/so/rules/nids/all.rules | wc -l
```

**Comment by gbrodt** [comment]
Yes, that is what is odd, no alerts at all  but the so-test has alerts show up.

The machines do have internet access.

The rules were updated.

There are 25365 rules active.

Greg

**Comment by dougburks** [comment]
When you import, is it generating logs and alerts in `/nsm/import/`?

What is the output of the following?
```
ls -R /nsm/import
```

**Reply by gbrodt** [reply]
![Pensession1-000240](https://user-images.githubusercontent.com/75701987/164317347-626413fd-ffc6-4d25-ad46-f8d6fb5b6b60.jpg)

If there is a way to private message you I can give you access to the pod.(username and password)

**Reply by dougburks** [reply]
Looking at your screenshot, it appears that Suricata is writing json files. You might look at those json files and see if they contain the alerts you expect. If so, you might check your Filebeat configuration to make sure it is consuming those files.

> If there is a way to private message you I can give you access to the pod.(username and password)

If that's the kind of support you're looking for, you might consider purchasing support:
https://securityonionsolutions.com/support/

**Comment by gbrodt** [comment]
Hi Doug,

Thank you I will take a look at them.

I offered the access as I thought you might want to see what was going on first hand. We use this to help teach students about cyber security in our program at the college and the support I suspect is out of our budget.  It's a great product and you have made great leaps forward with it - please keep up the great work.  Hopefully our grads will see your product thrive in industry.

I'll let you know if I find anything,
Greg

**Comment by gbrodt** [comment]
Update,  under nsm/import/(hashidofimport)/suricata 
The one json file is empty. No other info or logs under this.
The zeek and pcap subs have information in them. 

file beat log is showing: unable to load ingest pipelines for the configured modules because the elasticsearch output is not configured/enabled 
and
deprecated: log input use filestream input instead 


The /opt/so/log/suricata/import/{hashofinput}console.log seems to indicate issues with the input of the pcap
![Capture-000053](https://user-images.githubusercontent.com/75701987/165992624-6c9b140f-c19a-4e15-9c71-f083296e356b.png)

**Comment by dougburks** [comment]
![image](https://user-images.githubusercontent.com/1659467/166225150-74ebbc10-9984-484a-8656-57d84200230a.png)

Based on this error, have you checked the permissions on the pcap file that you're importing?

**Comment by gbrodt** [comment]
Permission on pcaps are 666  and I'm using the full path to the files on the import. Been looking at the import script and permissions but not finding an issue.  Could it be something with suricata running in user mode?

**Comment by dougburks** [comment]
> Could it be something with suricata running in user mode?

It's normal for Suricata to run in `USER` mode when reading a pcap file:
https://redmine.openinfosecfoundation.org/issues/2421

> Permission on pcaps are 666 and I'm using the full path to the files on the import.

Have you checked the permission on all of the parent directories in that full path?

Have you checked the SELinux context on the file?

Have you tried copying the file to `/tmp/` and importing from there?

**Reply by gbrodt** [reply]
Tried from tmp - same issue. I still need to look at context.

**Reply by dougburks** [reply]
@gbrodt Here's another test you can try:
- reboot the box
- login and run `sudo so-status` and wait for all services to show `OK` (if running in IMPORT node, the last 3 services may show as `DISABLED`)
- download zip file to /tmp:
```
cd /tmp
wget https://www.malware-traffic-analysis.net/2022/05/03/2022-05-03-Contact-Forms-Bumblebee-infection-with-Cobalt-Strike.pcap.zip
```
- extract zip file to obtain pcap file (enter password from https://www.malware-traffic-analysis.net/about.html):
```
unzip 2022-05-03-Contact-Forms-Bumblebee-infection-with-Cobalt-Strike.pcap.zip 
```
- import pcap file:
```
sudo so-import-pcap 2022-05-03-Contact-Forms-Bumblebee-infection-with-Cobalt-Strike.pcap
```
- wait 30 seconds for import to complete
- go to Alerts
- change timeframe so that the search includes 2022-05-03
- check to see if you now have alerts as shown at:
https://blogger.googleusercontent.com/img/b/R29vZ2xl/AVvXsEg0CnZBlx1SDkKsRvR_BcH7iLkT8zoup1OU5EnNncNRKAwK4UdXJOABphSNO5UroIKjwg1k3k-r50O1faVZVdu5PRH8vsHsDRMUtqUwZZyPiUUFKoBkgGkMVPEQaCS_-E0oXHdcr2F4UAviofpVQruSUoz9obY21rKEpLErPP8XEGe6qtm8BTS6fr9Iyg/s2536/Screen%20Shot%202022-05-05%20at%2011.58.34%20AM.png

**Reply by TotieBash** [reply]
Sorry I promise I will create my own thread.  I tested on two installs and it did not work.

SO 2.3.120
![image](https://user-images.githubusercontent.com/86748904/166984611-ea43e6cc-c18a-45c0-9e57-0d19ca4dbcf4.png)

![image](https://user-images.githubusercontent.com/86748904/166984729-b38e842a-9717-46ba-a15d-22f1c4ce9376.png)
SO 2.3.110
![image](https://user-images.githubusercontent.com/86748904/166984813-26ae9bba-be4d-4a8f-8c09-d87c788734c0.png)

![image](https://user-images.githubusercontent.com/86748904/166985010-94850ab4-7263-4f3f-a4b2-6ae2d1bba2a9.png)

**Reply by gbrodt** [reply]
Tag me in the tread when you create it as I would like to follow until a solution is found. Not sure why they want them separate when the issue is the same. Most discussion like to bring similar topics together.

**Reply by TotieBash** [reply]
I tagged you.  What was the last version that this so-import-pcap worked for you? 

https://github.com/Security-Onion-Solutions/securityonion/discussions/7906

**Reply by dougburks** [reply]
> Not sure why they want them separate when the issue is the same. Most discussion like to bring similar topics together.

@gbrodt This is our policy as listed at https://github.com/Security-Onion-Solutions/securityonion/discussions/1720. We have this policy because it is our experience that in many, if not most cases, folks may experience similar symptoms but have totally different root causes. Please try the test I mentioned above and share your results so that we can try to figure this out.

**Comment by TotieBash** [comment]
I have same issue in SO standalone iso install.  I tested on a new install of version 2.3.120.  I also tested on version  2.3.110.  I don't see any Suricata alerts on SOC and I do not see any in Kibana.  I tried the following three method
so-import-pcap 2022-03-01-Emotet-epoch4-infection-with-Cobalt-Strike-and-spambot-traffic.pcap
so-tcpreplay /opt/so/samples/2022-03-01-Emotet-epoch4-infection-with-Cobalt-Strike-and-spambot-traffic.pcap
tcpreplay -v --topspeed -i bond0 2022-03-01-Emotet-epoch4-infection-with-Cobalt-Strike-and-spambot-traffic.pcap

![image](https://user-images.githubusercontent.com/86748904/166846659-01acbf8c-e2ca-4271-a6b6-4e1b27330dbb.png)
![image](https://user-images.githubusercontent.com/86748904/166846793-539ce80e-0d40-430f-bf2f-8040218f01f7.png)

**Reply by dougburks** [reply]
@TotieBash From https://github.com/Security-Onion-Solutions/securityonion/discussions/1720:
> Start a new discussion instead of replying to somebody else's discussion
Please search to see if you can find similar discussions that may help you. However, in order to avoid confusion, please do NOT reply to somebody else's discussion with your own issue. Instead, please start a new discussion and in that new discussion you can provide a hyperlink to the related discussion.

**Comment by dougburks** [comment]
@gbrodt Have you had a chance to try the test I mentioned here?
https://github.com/Security-Onion-Solutions/securityonion/discussions/7786#discussioncomment-2695678

**Reply by gbrodt** [reply]
Sorry, been busy with students/finals. I was able to do this and I do not see the alert results as indicated in the link.
![Capture-000067](https://user-images.githubusercontent.com/75701987/168083678-5945cbc7-45d7-4860-95d3-b6ca33e12b8a.png)
![Capture-000066](https://user-images.githubusercontent.com/75701987/168083696-f7087166-9fc3-468d-a31b-8a5fcadb6657.png)

This is a standalone node with version 2.3.120 currently.

**Reply by gbrodt** [reply]
Also the hunt link shows the zeek events:
![Capture-000069](https://user-images.githubusercontent.com/75701987/168089211-ced49810-b856-4300-9730-f065778468d8.png)

**Comment by gbrodt** [comment]
SOLVED With help from my counterpart at Madison College Mike M. The root cause of the issue was the definition of the home/internal network. 
@TotieBash  @dougburks 

![Pensession1-000253](https://user-images.githubusercontent.com/75701987/168138797-6f2eea94-aa23-4b13-bbc8-b96a85684c3e.jpg)
![Pensession1-000254](https://user-images.githubusercontent.com/75701987/168138926-bbf06d18-a475-4306-b645-e283fccaa5d1.jpg)
![Pensession1-000255](https://user-images.githubusercontent.com/75701987/168139022-89f9263f-e6bd-47e5-9427-d0ccccac40de.jpg)

To fix it:

    sudo vi /opt/so/saltstack/local/pillar/global
edit the network number for the internal network.

sudo salt-call state.apply sutricata

Or use a tool like tracewranger and adjust address in the pcap to match your internal network definition! 

Lesson learned: Be sure your imports are using the same network address you defined for the internal network!

**Reply by gbrodt** [reply]
Second lesson/thought: Best practice to leave home networks at default RFC1918 numbers to detect and spoofed traffic.

**Reply by TotieBash** [reply]
ow!!  I guess this is the answer to my point #3, leave the $home_net as default.

**Comment by gbrodt** [comment]
Third thought: Do be lazy making your NAT - be specific on your home network so other numbers do not exit your network.

**Comment by TotieBash** [comment]
Thanks for the tag @gbrodt, I would have never found that, I'm not newbee with Suricata but that gotcha you guys found would have taken me a loooong time and that is if I am persistent to look for it.  So clarify something form me, and I know @dougburks will know the answer as well.  
1. Is there something that change from the old SO version to the new SO version, where Suricata config needed to be tweaked? I remember doing so-import-pcap on the old 2.3.60 version and it worked.
2. Did I get unlucky with the 3 pcap files that I tried and I just got lucky with my last years so-import-pcap?
3. Is the fix related to this install section?  I answer 10.239.1.0/24 CIDR block for my network.  It looks like I should have left this section as default with the entire RFC1918 (this is assuming that this section goes to the Suricata config).
![image](https://user-images.githubusercontent.com/86748904/168148643-b5ded860-7365-420c-88a4-b52359771567.png)
4. May be for @dougburks, forgive me for assuming, if its not in so-import-pcap logic to check Suricata $home_net or maybe throw an error code at least.  
5. This explains why so-import-pcap, so-tcpreplay and vanilla tcpreplay did not throw any Suricata alerts for me.
6. @gbrodt your second screenshot, which Suricata log file location is that?

Thanks again for the fix, I will try it on my lab as soon as I get my hands on it next week.  But I already know it will work.  thanks again..

**Reply by dougburks** [reply]
> 1. Is there something that change from the old SO version to the new SO version, where Suricata config needed to be tweaked? I remember doing so-import-pcap on the old 2.3.60 version and it worked.

Security Onion has defaulted the homenet value to RFC1918 private address space for many years now, possibly since the beginning of the project in 2008.

> 2\. Did I get unlucky with the 3 pcap files that I tried and I just got lucky with my last years so-import-pcap?

I suspect that your installation last year had the default homenet value of RFC1918 private address space.

> 3\. Is the fix related to this install section?  I answer 10.239.1.0/24 CIDR block for my network.  It looks like I should have left this section as default with the entire RFC1918 (this is assuming that this section goes to the Suricata config).

Yes, it's related to that question and the answer goes into the Suricata config. If you're only ever monitoring 10.239.1.0/24, then you can restrict it to that. But if you ever want to replay or import pcaps from another network, you will either need to change that value or not get alerts.

> 4\. May be for @dougburks, forgive me for assuming, if its not in so-import-pcap logic to check Suricata $home_net or maybe throw an error code at least.

so-import-pcap has no way of knowing if you set your homenet value correctly.

**Reply by TotieBash** [reply]
Ok thanks for the reply.  Very helpful info.

**Comment by dougburks** [comment, marked as answer]
Security Onion defaults the homenet value to RFC1918 private address space. You can see this in the screenshots at https://docs.securityonion.net/en/2.3/first-time-users.html:

![image](https://user-images.githubusercontent.com/1659467/168265932-63a6c24e-297b-462b-a108-76488c0b3e51.png)

Changing this default setting changes the networks that Suricata and Zeek will monitor.

I've added a new Troubleshooting Alerts section to the Suricata page of the documentation:
https://docs.securityonion.net/en/2.3/suricata.html#troubleshooting-alerts

I've also updated https://docs.securityonion.net/en/2.3/homenet.html.
